File: tests/test_foMPI/win_name.c
/* This test is derived from the win_name test of the RMA tests of mpich-1.5b2. */

#include <stdio.h>
#include <string.h>

#include "fompi.h"
#include "test_mpi3rma.h"

static int cnt = 0;

void test_mpi3rma_win_name( foMPI_Win win ) {

  char name[MPI_MAX_OBJECT_NAME];
  char getname[MPI_MAX_OBJECT_NAME];
  int length;
  int status;

  snprintf(name, MPI_MAX_OBJECT_NAME, "win-%d", cnt++);

  status = foMPI_Win_set_name( win, &name[0] );
  _check_fompi_status(status, MPI_SUCCESS, (char*) __FILE__, __LINE__);
  status = foMPI_Win_get_name( win, &getname[0], &length );
  _check_fompi_status(status, MPI_SUCCESS, (char*) __FILE__, __LINE__);

  if( strncmp(name, getname, MPI_MAX_OBJECT_NAME) == 0 ) {
    printf("No Error\n");
  } else {
    printf("Error: Got name %s instead of %s.\n", getname, name);
  }

  return;
}

File: tests/test_mpi3rma_fortran/win_name.f90
! This test is derived from the win_name test of the RMA tests of mpich-1.5b2.

      subroutine test_mpi3rma_win_name( win )

      use utilities, only: check_fompi_status

#if INCLUDE_TYPE==1
      implicit none

      include 'fompif.h'
#endif
#if INCLUDE_TYPE==2
      use fompi

      implicit none
#endif

      integer, intent(in) :: win

      character(MPI_MAX_OBJECT_NAME) :: window_name
      character(MPI_MAX_OBJECT_NAME) :: get_window_name
      integer :: length
      integer :: mpierr
      integer :: cnt = 0

      character(50) :: filename

      write (filename,*) "win_name.f90"

      write (window_name,*) "win-", cnt
      cnt = cnt + 1

      call foMPI_Win_set_name( win, window_name, mpierr )
      call check_fompi_status(mpierr, MPI_SUCCESS, filename, 27, 0 ) ! only rank 0 calls this test routine

      call foMPI_Win_get_name( win, get_window_name, length, mpierr )
      call check_fompi_status(mpierr, MPI_SUCCESS, filename, 30, 0 ) ! only rank 0 calls this test routine

      if( window_name .eq. get_window_name ) then
        write (*,*) "No Error"
      else
        write (*,*) "Error: Got name ", trim(get_window_name), " instead of ", &
          trim(window_name), "."
      endif

      end subroutine test_mpi3rma_win_name
